File: Python/2601-2650/2610. Convert an Array Into a 2D Array With Conditions.py
# https://leetcode.com/problems/convert-an-array-into-a-2d-array-with-conditions/

# https://leetcode.com/problems/convert-an-array-into-a-2d-array-with-conditions/solutions/4490403/o-n-java-python-javascript-linear-solution/

class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:

        frequency = {}
        result = []

        for num in nums:
            if num not in frequency:
                frequency[num] = 0
            if frequency[num] >= len(result): # Create new Row
                result.append([])
            
            result[frequency[num]].append(num)
            frequency[num] += 1
        
        return result

File: Java/2601-2650/2610. Convert an Array Into a 2D Array With Conditions.java
// https://leetcode.com/problems/convert-an-array-into-a-2d-array-with-conditions/

// https://leetcode.com/problems/convert-an-array-into-a-2d-array-with-conditions/solutions/4490403/o-n-java-python-javascript-linear-solution/

class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {

        int[] frequency = new int[nums.length+1];
        ArrayList<List<Integer>> result = new ArrayList<>();

        for(int num : nums){
            if(frequency[num] >= result.size()) // Create new Row
                result.add(new ArrayList<>());
            
            result.get(frequency[num]++).add(num);
        }

        return result;
    }
}
